pkg/network/ebpf/c/prebuilt/http.c

```diff
@@ -225,6 +225,107 @@ int uretprobe__SSL_write(struct pt_regs* ctx) {
     return 0;
 }
 
+SEC("uprobe/SSL_read_ex")
+int uprobe__SSL_read_ex(struct pt_regs* ctx) {
+    ssl_read_ex_args_t args = {0};
+    args.ctx = (void *)PT_REGS_PARM1(ctx);
+    args.buf = (void *)PT_REGS_PARM2(ctx);
+    args.size_out_param = (size_t *)PT_REGS_PARM4(ctx);
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    log_debug("uprobe/SSL_read_ex: pid_tgid=%llx ctx=%llx\n", pid_tgid, args.ctx);
+    bpf_map_update_elem(&ssl_read_ex_args, &pid_tgid, &args, BPF_ANY);
+    return 0;
+}
+
+SEC("uretprobe/SSL_read_ex")
+int uretprobe__SSL_read_ex(struct pt_regs* ctx) {
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    int return_code = (int)PT_REGS_RC(ctx);
```

Comment:
Why not check the return code as first step? do we really need this log message?

Reply:
it is not for the log, but for the cleanup